packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -726,6 +727,11 @@ function ensureRootIsScheduled(root: FiberRoot, currentTime: number) {
     newCallbackNode = scheduleSyncCallback(
       performSyncWorkOnRoot.bind(null, root),
     );
+  } else if (newCallbackPriority === SyncBatchedLanePriority) {
+    newCallbackNode = scheduleCallback(
```

Comment:
Are we intentionally not using `scheduleSyncCallback` like in the block above?

What if other sync work is already scheduled, do we want this to always come after?

Reply:
The naming is confusing. `scheduleSyncCallback` adds stuff to a special queue that can be flushed synchronously without going through Scheduler. It's used for SyncLane and expired updates.

"sync" in `performSyncWorkOnRoot` means that it doesn't yield to the main thread and it doesn't check Suspense timeouts (though that part might change), so it has fewer checks than concurrent updates.

SyncBatched is kind of a mix  between sync mode and concurrent mode. It doesn't yield to the main thread, but it does go through Scheduler (i.e. is scheduled by postMessage), hence the weird naming conflicts.

Reply:
I see, thanks for explaining 👍